### PR TITLE
Add support for curve ed25519

### PR DESCRIPTION
**Pull request by matthiasgeihs**
This adds support for ed25519 curve.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: CI
```
[the remark this answers is not in the document]
Comment (matthiasgeihs) [reply .github/workflows/ci.yaml:1]: Welcome :)

**Comment by matthiasgeihs** [comment]
Need to bump version? @himanshuchawla009

**Review by himanshuchawla009 — APPROVED** [review]

**Comment by himanshuchawla009** [comment]
> Need to bump version? @himanshuchawla009

i will bump it during release, it doesnt seems to have any breaking versions right, all apis and response types are backward compatible?

**Comment by matthiasgeihs** [comment]
> > Need to bump version? @himanshuchawla009
> 
> i will bump it during release, it doesnt seems to have any breaking versions right, all apis and response types are backward compatible?

Technically, there are some breaking changes:
1. The client now expects a `key_type` parameter upon initialization.
2. The client will now send a `key_type` parameter with every request to the server.

**Comment by himanshuchawla009** [comment]
> > > Need to bump version? @himanshuchawla009
> > 
> > 
> > i will bump it during release, it doesnt seems to have any breaking versions right, all apis and response types are backward compatible?
> 
> Technically, there are some breaking changes:
> 
>     1. The client now expects a `keyType` parameter upon initialization.
> 
>     2. The client will now send a `key_type` parameter with every request to the server.
> 
> 
> (1.) could be made non-breaking by using `secp256k1` as default if not `keyType` is provided. (2.) may be breaking or may not be depending on how the server handles unknown parameters: does it throw? or just ignore?

Just tested, server doesn't throws currently on passing unknown params.

**Comment by matthiasgeihs** [comment]
So does this PR look good? Or you want me to change anything? @himanshuchawla009

**Comment by matthiasgeihs** [comment]
@himanshuchawla009, have created alpha branch and targeted that one here. Please merge and publish when you are ready. Also, could you please set alpha branch as default?

**Pull request merged**